core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -562,6 +578,28 @@ public synchronized WorkerNetAddress getLocalWorker() throws IOException {
     return mLocalWorker;
   }
 
+  /**
+   * @return the info of all block workers eligible for reads and writes
+   */
+  public synchronized List<BlockWorkerInfo> getEligibleWorkers() throws IOException {
```

Comment:
It seems like the only difference between this method and `getAllWorkers` is that this one is cached (so maybe stale)? It is not clear from the method names, and or java doc.

Reply:
Agree. I would rename this to `getCachedWorkers`